packages/walletconnect-v2/src/index.ts

```diff
@@ -86,27 +111,47 @@ export class WalletConnect extends Connector {
     this.events.emit(URI_AVAILABLE, uri)
   }
 
+  private async initializeProvider(
+    desiredChainId: number | undefined = this.defaultChainId
+  ): Promise<WalletConnectProvider> {
+    const ethProviderModule = await import('@walletconnect/ethereum-provider')
+    const rpcMap = this.rpcMap ? await getBestUrlMap(this.rpcMap, this.timeout) : undefined
+    const chainProps = this.getChainProps(this.chains, this.optionalChains, desiredChainId)
+
+    this.provider = await ethProviderModule.default.init({
+      ...this.options,
+      ...chainProps,
+      rpcMap,
+    })
+
+    return this.handleProviderEvents(this.provider)
```

Comment:
You've introduced a waterfall pattern by modifying the rpcMap initialization:

```suggestion
    const rpcMap = this.rpcMap ? getBestUrlMap(this.rpcMap, this.timeout) : undefined
    const chainProps = this.getChainProps(this.chains, this.optionalChains, desiredChainId)

    this.provider = (await import('@walletconnect/ethereum-provider')).default.init({
      ...this.options,
      ...chainProps,
      rpcMap: await rpcMap,
    })

    return this.handleProviderEvents(this.provider)
```

Note that with this change, `rpcMap` can resolve while the provider module loads.

Reply:
This.provider give the following error w/ this pattern

```Type 'Promise<EthereumProvider>' is missing the following properties from type 'EthereumProvider': events, namespace, accounts, signer, and 35 more.ts(2740)```